packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
+            defaultGCData.gcNodes[nodes[2]] = [ nodes[3] ];
+            defaultGCData.gcNodes[nodes[3]] = [ nodes[0] ];
+        });
 
-    it("doesn't generate inactive events for referenced nodes", async () => {
-        const garbageCollector = createGarbageCollector();
+        it("doesn't generate inactive events for referenced nodes", async () => {
+            const garbageCollector = createGarbageCollector();
 
-        // Run garbage collection on the default GC data where everything is referenced.
-        await garbageCollector.collectGarbage({ runGC: true });
+            // Run garbage collection on the default GC data where everything is referenced.
+            await garbageCollector.collectGarbage({ runGC: true });
 
-        // Change all nodes.
-        changeAllNodes(garbageCollector);
+            // Change all nodes.
+            changeAllNodes(garbageCollector);
 
-        // Validate that no inactive events are generated yet.
-        validateNoInactiveEvents();
+            // Validate that no inactive events are generated yet.
+            validateNoInactiveEvents();
 
-        // Wait for unreferenced timer (if any) to expire.
-        await waitForDeleteTimeout();
+            // Wait for unreferenced timer (if any) to expire.
+            await waitForDeleteTimeout();
 
-        // Change all nodes again.
-        changeAllNodes(garbageCollector);
+            // Change all nodes again.
+            changeAllNodes(garbageCollector);
 
-        // Validate that no inactive events are generated since everything is referenced.
-        validateNoInactiveEvents();
-    });
+            // Validate that no inactive events are generated since everything is referenced.
+            validateNoInactiveEvents();
+        });
 
-    it("generates inactive events when inactive node is changed or revived", async () => {
-        const garbageCollector = createGarbageCollector();
+        it("generates inactive events when inactive node is changed or revived", async () => {
+            const garbageCollector = createGarbageCollector();
+
+            // Remove node 2's reference from node 1. This should make node 2 and node 3 unreferenced.
+            defaultGCData.gcNodes[nodes[1]] = [];
+
+            await garbageCollector.collectGarbage({ runGC: true });
+
+            // Change all nodes.
+            changeAllNodes(garbageCollector);
+
+            // Validate that no inactive events are generated yet.
+            validateNoInactiveEvents();
+
+            // Wait for unreferenced timer (if any) to expire.
+            await waitForDeleteTimeout();
+
+            // Change all nodes. This should result in an inactiveObjectChanged event for node 2 and node 3 since they
+            // are inactive.
+            changeAllNodes(garbageCollector);
+            assert(
+                mockLogger.matchEvents([
+                    { eventName: inactiveObjectChangedEvent, deleteTimeoutMs, inactiveNodeId: nodes[2] },
```

Comment:
I'd prefer for us to use smaller set of fields/properties in telemetry, as it otherwise explodes, and affects storage cost, but also ease of authoring simple queries. Can we use 'duration' ?

Reply:
Somewhat similar for inactiveNodeId - can it be `id`?